modules/sdk-coin-opeth/test/unit/opeth.ts

```diff
@@ -778,5 +781,68 @@ describe('Optimism', function () {
       transaction.halfSigned?.should.have.property('txHex');
       transaction.halfSigned?.should.have.property('recipients');
     });
+
+    it('should generate an half signed recovery txn of a token for hot wallet', async function () {
+      const userKey =
+        '{"iv":"VFZ3jvXhxo1Z+Yaf2MtZnA==","v":1,"iter":10000,"ks":256,"ts":64,"mode"\n' +
+        ':"ccm","adata":"","cipher":"aes","salt":"p+fkHuLa/8k=","ct":"hYG7pvljLIgCjZ\n' +
+        '53PBlCde5KZRmlUKKHLtDMk+HJfuU46hW+x+C9WsIAO4gFPnTCvFVmQ8x7czCtcNFub5AO2otOG\n' +
+        'OsX4GE2gXOEmCl1TpWwwNhm7yMUjGJUpgW6ZZgXSXdDitSKi4V/hk78SGSzjFOBSPYRa6I="}\n';
+      const walletContractAddress = TestBitGo.V2.TEST_ETH_WALLET_FIRST_ADDRESS as string;
+      const walletPassphrase = TestBitGo.V2.TEST_RECOVERY_PASSCODE as string;
+      const tokenContractAddress = '0xe4ab69c077896252fafbd49efd26b5d171a32410'; // topeth-link contract token address
+
+      const basecoin = bitgo.coin('topeth') as Opeth;
+      nock(baseUrl)
+        .get('/api')
+        .query(mockData.getTxListRequest(bitgoFeeAddress))
+        .reply(200, mockData.getTxListResponse);
+      nock(baseUrl)
+        .get('/api')
+        .query(mockData.getBalanceRequest(bitgoFeeAddress))
+        .reply(200, mockData.getBalanceResponse);
+      nock(baseUrl)
+        .get('/api')
+        .query(mockData.getTokenBalanceRequest(tokenContractAddress, walletContractAddress))
+        .reply(200, mockData.getBalanceResponse);
+      nock(baseUrl).get('/api').query(mockData.getContractCallRequest).reply(200, mockData.getContractCallResponse);
+
+      const transaction = (await basecoin.recover({
+        userKey: userKey,
+        backupKey: '',
+        walletPassphrase: walletPassphrase,
+        walletContractAddress: walletContractAddress,
+        bitgoFeeAddress: bitgoFeeAddress,
+        recoveryDestination: destinationAddress,
+        eip1559: { maxFeePerGas: 20000000000, maxPriorityFeePerGas: 10000000000 },
+        gasLimit: 500000,
+        bitgoDestinationAddress: bitgoDestinationAddress,
+        tokenContractAddress: tokenContractAddress,
+      })) as OfflineVaultTxInfo;
+
+      should.exist(transaction);
+      transaction.should.have.property('txHex');
```

Comment:
build the transaction from txHex and see whether tokenContractAddress is defined

Reply:
built tx from txHex and added check that data should start with `0x0dcd7a6c`